src/pru-motion-queue.cc

```diff
@@ -216,5 +177,10 @@ bool PRUMotionQueue::Init() {
   }
   queue_pos_ = 0;
 
+  // The MSB 8 bits of the status register shoud represent the index
+  // of the last executed motion segment. If no motion segments has been
+  // yet pushed in the queue, the index should represent the index before
+  // the slow we would push, in this case -1.
```

Comment:
so, uhm, why not call it -1 then instead of 0xff ? If status.index is a char, it should set it to the same bit-pattern.

I'd probably not use this constant here, but have some
  constexpr int kInitializedIndexSentinel = 0xff;
or something. And then use this here and in the Status() function where you look at it.